taxi/tests.py

```diff
@@ -1,3 +1,108 @@
 from django.test import TestCase
+from taxi.forms import validate_license_number, ManufacturerSearchByName
+from django.core.exceptions import ValidationError
+from parameterized import parameterized
+from taxi.models import Driver, Manufacturer, Car
+from django.urls import reverse
 
-# Create your tests here.
+
+class FormTest(TestCase):
+    def test_license_is_valid(self):
+        license_number = "ABC12345"
+        req = validate_license_number(license_number)
+        self.assertEqual(req, license_number)
+
+    @parameterized.expand([
+        ("Test length of license number", "ABC1234"),
+        ("Test first 3 letters is upper", "abc12345"),
+        ("Test first 3 characters is letter", "AB123456"),
+        ("Test last 5 characters is digits", "ABCD1234")
+    ])
+    def test_license_is_invalid(self, test_name: str, number: str):
+        license_number = number
+        with self.assertRaises(ValidationError, msg=test_name):
+            validate_license_number(license_number)
+
+
+class ViewTest(TestCase):
+    def setUp(self):
+        self.driver1 = Driver.objects.create_user(
+            username="test_user",
+            password="test_pass",
+            license_number="ABC12345"
+        )
+        self.driver2 = Driver.objects.create(
+            username="Alex",
+            password="test_pass",
+            license_number="ABC23456"
+        )
+        self.factory1 = Manufacturer.objects.create(
+            name="BMW",
+            country="Germany"
+        )
+        self.factory2 = Manufacturer.objects.create(
+            name="Mercedes",
+            country="Germany"
+        )
+        self.factory3 = Manufacturer.objects.create(
+            name="Toyota",
+            country="Japan"
+        )
+        self.car1 = Car.objects.create(
+            model="Prius",
+            manufacturer=self.factory3,
+        )
+        self.car2 = Car.objects.create(
+            model="M3",
+            manufacturer=self.factory2,
+        )
+
+    list_and_create_urls = [
+        "taxi:index",
+        "taxi:manufacturer-list",
+        "taxi:manufacturer-create",
+        "taxi:car-list",
+        "taxi:car-create",
+        "taxi:driver-list",
+        "taxi:driver-create",
+    ]
+    update_and_delete_urls = [
+        ("taxi:manufacturer-update", 1),
+        ("taxi:manufacturer-delete", 1),
+        ("taxi:car-update", 1),
+        ("taxi:car-delete", 1),
+        ("taxi:driver-update", 1),
+        ("taxi:driver-delete", 1)
+    ]
+
+    @parameterized.expand(list_and_create_urls)
+    def test_authenticated_user(self, view_name):
```

Comment:
Test name is not clear. You should name it according to a feature you want to test. Looks now like you want to test if user can be authenticated